questionable/private/binderror.nim

```diff
@@ -1,24 +1,43 @@
 import std/options
+import std/macros
 
-var captures {.global, compileTime.}: int
-var errorVariable {.threadvar.}: ptr ref CatchableError
+# A stack of names of error variables. Keeps track of the error variables that
+# are given to captureBindError().
+var errorVariableNames {.global, compileTime.}: seq[string]
```

Comment:
I like this scheme better than the global var. Hopefully it isolates the error variables better too.